code/modules/telesci/telesci_computer.dm

```diff
@@ -390,8 +415,42 @@
 		eject()
 		temp_msg = "NOTICE:<BR>Bluespace crystals ejected."
 
+	if(href_list["toggle_device_mode"])
+		device_mode = !device_mode
+
+	if(href_list["edit_device"])
+		if(attached_device)
+			attached_device.attack_self(usr)
+	if(href_list["remove_device"])
+		attached_device.loc = get_turf(src)
+		attached_device:holder = null
+		attached_device = null
+
 	updateUsrDialog()
 
+/obj/machinery/computer/telescience/HasProximity(atom/movable/AM as mob|obj)
+	if(!attached_device)	return
+	attached_device.HasProximity(AM)
+	return
+
+/obj/machinery/computer/telescience/hear_talk(mob/living/M as mob, msg)
+	..()
+	for(var/obj/O in contents)
+		O.hear_talk(M, msg)
+
+/obj/machinery/computer/telescience/hear_message(mob/living/M as mob, msg)
+	..()
+	for(var/obj/O in contents)
+		O.hear_message(M, msg)
+
+/obj/machinery/computer/telescience/proc/process_activation(var/obj/item/device/D)
+	if(toggle)
+		toggle = 0
```

Comment:
You should use a more clear variable name, such as `cooldown` or some such, and blindly spawn()'ing for long periods of time isn't ideal; It should probably be more like
```
#define REMOTE_COOLDOWN 50
/obj/machinery/computer/telescience
    var/remote_cooldown = 0

/obj/machinery/computer/telescience/proc/process_activation(var/obj/item/device/D)
    if(tele_cooldown > world.time)
        return 0
    remote_cooldown = world.time + REMOTE_COOLDOWN
    sending = device_mode
    teleport(usr)
```